packages/dd-trace/test/config.spec.js

```diff
@@ -397,6 +412,38 @@ describe('Config', () => {
     expect(config).to.have.property('plugins', false)
   })
 
+  it('should initialize from the options with simplified trace propagation style', () => {
+    const config = new Config({
+      tracePropagationStyle: 'datadog'
+    })
+
+    expect(config).to.have.nested.deep.property('tracePropagationStyle.inject', ['datadog'])
+    expect(config).to.have.nested.deep.property('tracePropagationStyle.extract', ['datadog'])
+  })
+
+  it('should warn if mixing propagation style env vars', () => {
+    const expected = 'Use either the DD_TRACE_PROPAGATION_STYLE environment variable or ' +
+      'separate DD_TRACE_PROPAGATION_STYLE_INJECT and DD_TRACE_PROPAGATION_STYLE_EXTRACT' +
+      'environment variables'
+
+    process.env.DD_TRACE_PROPAGATION_STYLE_EXTRACT = 'datadog'
+    process.env.DD_TRACE_PROPAGATION_STYLE = 'datadog'
+
+    // eslint-disable-next-line no-new
+    new Config()
+
+    expect(log.warn).to.have.been.calledWith(expected)
+
+    delete process.env.DD_TRACE_PROPAGATION_STYLE_EXTRACT
```

Comment:
These will never happen if the assertion fails. It would be safer to reset the values in an `afterEach`.